src/ir_Technibel.cpp

```diff
@@ -0,0 +1,416 @@
+//            Typically/Defaults to kStartOffset.
+//   nbits:   The number of data bits to expect. Typically kTechnibelAcBits.
+//   strict:  Flag indicating if we should perform strict matching.
+// Returns:
+//   boolean: True if it can decode it, false if it can't.
+//
+// Status: STABLE / Expected to be working.
+bool IRrecv::decodeTechnibelAc(decode_results *results, uint16_t offset,
+                              const uint16_t nbits, const bool strict) {
+  if (results->rawlen < 2 * nbits + kTechnibelAcOverhead - offset) {
+    return false;  // Too short a message to match.
+  }
+  if (strict && nbits != kTechnibelAcBits) {
+    return false;
+  }
+
+  uint64_t data = 0;
+
+  // Header + Data + Footer
+  if (!matchGeneric(results->rawbuf + offset, &data,
+                    results->rawlen - offset, nbits,
+                    kTechnibelAcHdrMark, kTechnibelAcHdrSpace,
+                    kTechnibelAcBitMark, kTechnibelAcOneSpace,
+                    kTechnibelAcBitMark, kTechnibelAcZeroSpace,
+                    kTechnibelAcBitMark, kTechnibelAcGap, true,
+                    _tolerance, kMarkExcess, false)) return false;
+
+  // Success
+  results->decode_type = decode_type_t::TECHNIBEL_AC;
+  results->bits = nbits;
+  results->value = data;
+  results->command = 0;
+  results->address = 0;
+  return true;
+}
+#endif  // DECODE_TECHNIBEL_AC
+
+
+// Class for controlling the settings of a Technibel A/C
+
+IRTechnibelAc::IRTechnibelAc(const uint16_t pin, const bool inverted,
+                           const bool use_modulation)
+      : _irsend(pin, inverted, use_modulation) { this->stateReset(); }
+
+void IRTechnibelAc::begin(void) { _irsend.begin(); }
+
+#if SEND_TECHNIBEL_AC
+void IRTechnibelAc::send(const uint16_t repeat) {
+  _irsend.sendTechnibelAc(getRaw(), kTechnibelAcBits, repeat);
+}
+#endif  // SEND_TECHNIBEL_AC
+
+uint8_t IRTechnibelAc::calcChecksum(const uint64_t state) {
+  uint8_t sum = 0;
+  // Add up all the 8 bit data chunks.
+  for (uint8_t offset = kTechnibelAcPowerBit;
+        offset < kTechnibelAcFooterOffset; offset += 8) {
+    sum += reverseBits(GETBITS64(state, offset, 8), 8);
+  }
+  sum -= 1;
+  sum = invertBits(sum, 8);
+  sum = reverseBits(sum, 8);
+
+  return sum;
+}
+
+bool IRTechnibelAc::validChecksum(const uint64_t state) {
+  return (GETBITS64(state, kTechnibelAcChecksumOffset,
+                    kTechnibelAcChecksumSize) ==
+      IRTechnibelAc::calcChecksum(state));
+}
+
+void IRTechnibelAc::checksum(void) {
+  setBits(&remote_state, kTechnibelAcChecksumOffset, kTechnibelAcChecksumSize,
+          calcChecksum(remote_state));
+}
+
+void IRTechnibelAc::stateReset(void) {
+  remote_state = 0x00000000000018;
+  _saved_temp = 20;  // DegC  (Random reasonable default value)
+  _saved_temp_units = 0;  // Celsius
+
+  this->off();
+  this->setTemp(_saved_temp);
+  this->setTempUnit(_saved_temp_units);
+  this->setMode(kTechnibelAcCool);
+  this->setFan(kTechnibelAcFanLow);
```

Comment:
Please drop the use of `this->` unless it is really needed anywhere in the code. It's a style decision we came up with only recently. We are slowly removing it from all the existing code. So might as well not add more instances of it.